knative-operator/pkg/controller/knativeserving/knativeserving_controller.go

```diff
@@ -51,6 +57,40 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 		return err
 	}
 
+	resources, err := kourier.Resources(ingressNamespace)
+	if err != nil {
+		return err
+	}
+	// Watch Kourier resources.
+	for _, u := range resources {
```

Comment:
I **think** we might need to do some shuffling of the data first. A `Watch` watches all resources of the specified type. I think we want to watch all the relevant types and then filter by the resources we actually know we want to watch.

To do that, I think we need to build a `map[schema.GroupVersionKind][]types.NamespacedName`, then loop over all keys, setup a watch for each of the GroupVersionKinds and inside the `ToRequests` func, make sure that the current object matches one of the namespaced names in the list of the relevant GVK.